src/Tickets/Commerce/Gateways/Square/REST/Webhook_Endpoint.php

```diff
@@ -202,19 +208,46 @@ public function handle_webhook( WP_REST_Request $request ) {
 	protected function process_webhook_event( array $event_data ) {
 		$event_type = $event_data['type'] ?? '';
 
+		if ( ! in_array( $event_type, Events::get_webhook_event_types(), true ) ) {
+			do_action(
+				'tribe_log',
+				'warning',
+				'Unsupported Square webhook event type',
+				[
+					'source'     => 'tickets-commerce-square',
+					'event_type' => $event_type,
+					'data'       => $event_data,
+				]
+			);
+			return;
+		}
+
+		$event_mode = $event_data['env'] ?? 'sandbox';
+		$callback   = fn() => 'sandbox' === $event_mode;
+
+		add_filter( 'tec_tickets_commerce_is_sandbox_mode', $callback );
```

Comment:
Should there be here?

Reply:
Yes, thats how while processing the webhook all the other functions react like the current mode is in the same mode as the incoming event's mode.

Reply:
removed that as we discussed. Now we wont handle events for an env we are not currently at ...